libs/experimental/langchain_experimental/tools/cpp/tool.py

```diff
@@ -0,0 +1,134 @@
+import os
+import re
+import subprocess
+import tempfile
+from typing import Any, Optional, Type
+
+from langchain.tools.base import BaseTool
+from langchain_core.callbacks.manager import (
+    AsyncCallbackManagerForToolRun,
+    CallbackManagerForToolRun,
+)
+from langchain_core.runnables.config import run_in_executor
+from pydantic import BaseModel, Field
+
+
+def sanitize_cpp_input(code: str) -> str:
+    """Sanitize input for subprocess
+
+    Remove unnecessary whitespace and backticks
+
+    Args:
+        code: The code to sanitize
+
+    Returns:
+        str: The sanitized code
+    """
+    # Removes backticks and leading/trailing whitespace
+    code = re.sub(r"`", "", code)
+    code = code.strip()
+    return code
+
+
+class CppInputs(BaseModel):
+    """C/C++ inputs"""
+
+    code: str = Field(description="C/C++ code snippet to run")
+    std: str = Field(
+        default="c++14",
+        description="C++ standard to use (e.g., c++11, c++14, c++17, c++20)",
+    )
+    language: str = Field(
+        default="cpp", description="Programming language (either 'c' or 'cpp')"
+    )
+    cpu_limit: Optional[int] = Field(
+        default=None, description="Optional CPU time limit in seconds"
+    )
+
+
+class CppSubprocessTool(BaseTool):
+    """Tool for running C/C++ code using subprocess and g++/gcc"""
+
+    name: str = "SubprocessCpp"
+    description: str = (
+        "A C/C++ interpreter using subprocess and g++/gcc. "
+        "Use this to execute C/C++ commands. "
+        "Input should be a valid C/C++ code snippet. "
+        "If you want to see the output of a value, you should print it out"
+        "with `std::cout << ...` or `printf(...)`."
+    )
+    args_schema: Type[BaseModel] = CppInputs  # type: ignore
+    sanitize_input: bool = True
```

Comment:
Major: Could you document what sanitize input means? 

The variable name makes it sound like it sandboxes the execution of the code, which it does not.

Reply:
Resolved in the next commit, documented in tool.